### ISSUE TITLE
Output of sysconfig.get_link_flags does not seem to be suitable for Mac

### ISSUE THREAD

**Issue by pio-neil** [tensorflow/tensorflow#30564]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac High Sierra 10.13.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): v1.14.0-rc1-22-gaf24dc91b5 1.14.0
- Python version: Python 3.6.6 (v3.6.6:4cf1f54eb7, Jun 26 2018, 19:50:54)
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): Apple LLVM version 9.0.0 (clang-900.0.39.2)
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

When running the example of creating a custom op from https://www.tensorflow.org/guide/extend/op on a Mac the compilation stage (see (https://www.tensorflow.org/guide/extend/op#build_the_op_library) fails with the following error message:

```
ld: library not found for -l:libtensorflow_framework.1.dylib
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

This appears to be because the specified linker option `-l:libtensorflow_framework.1.dylib` (returned by `sysconfig.get_link_flags`) is not valid for `ld` on the Mac. The man page for `ld` says the following:

```
-lx         This option tells the linker to search for libx.dylib or libx.a in the library 
search path.  If string x is of the form y.o, then that file is searched for in the 
same places, but without prepending `lib' or appending `.a' or `.dylib' to the 
filename.
```

I believe that the correct format for this flag is `-ltensorflow_framework.1` (since the linker will prepend `lib` and append `.dylib`).

A workaround for this is to replace this line

```
TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
```

With this:

```
TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()).replace("-l:libtensorflow_framework.1.dylib", "-ltensorflow_framework.1"))') )
```

Alternatively there is (I believe) a fix for the underlying cause here: https://github.com/pio-neil/tensorflow/pull/1/files

**Describe the expected behavior**

The linking process should be performed successfully and produce a library file called zero_out.so.

**Code to reproduce the issue**

On a Mac, create a file called zero_out.cc with the following code (copied from https://www.tensorflow.org/guide/extend/op):

```
#include "tensorflow/core/framework/op.h"
#include "tensorflow/core/framework/shape_inference.h"
#include "tensorflow/core/framework/op_kernel.h"


using namespace tensorflow;

REGISTER_OP("ZeroOut")
    .Input("to_zero: int32")
    .Output("zeroed: int32")
    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
      c->set_output(0, c->input(0));
      return Status::OK();
    });

class ZeroOutOp : public OpKernel {
 public:
  explicit ZeroOutOp(OpKernelConstruction* context) : OpKernel(context) {}

  void Compute(OpKernelContext* context) override {
    // Grab the input tensor
    const Tensor& input_tensor = context->input(0);
    auto input = input_tensor.flat<int32>();

    // Create an output tensor
    Tensor* output_tensor = NULL;
    OP_REQUIRES_OK(context, context->allocate_output(0, input_tensor.shape(),
                                                     &output_tensor));
    auto output_flat = output_tensor->flat<int32>();

    // Set all but the first element of the output tensor to 0.
    const int N = input.size();
    for (int i = 1; i < N; i++) {
      output_flat(i) = 0;
    }

    // Preserve the first input value if possible.
    if (N > 0) output_flat(0) = input(0);
  }
};

REGISTER_KERNEL_BUILDER(Name("ZeroOut").Device(DEVICE_CPU), ZeroOutOp);
```

Then run:

```
TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
g++ -std=c++11 -shared zero_out.cc -o zero_out.so -fPIC ${TF_CFLAGS[@]} ${TF_LFLAGS[@]} -O2
```

You should see an error like this:

```
ld: library not found for -l:libtensorflow_framework.1.dylib
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by rrricharrrd** [comment]
I have seen the same issue. Suggested fix works for me

**Comment by yongtang** [comment]
The issue has been fixed through PR #30656
